spec/helpers.lua

```diff
@@ -186,9 +296,167 @@ Status received:
+-- Parameter name comparison is done case-insensitive.
+-- @return value of the parameter
+local function req_query_param(state, args)
+  local param = args[1]
+  local req = args[2] or kong_state.kong_request
+  if (not req) and kong_state.kong_response then
+    table.insert(args, 1, "<< 'queryparam' assertion only works with a request object >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  if (type(req) ~= "table") or (type(req.queryString) ~= "table") then
+    table.insert(args, 1, "<< 'queryparam' assertion input does not contain a 'queryString' subtable >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  local value = lookup(req.queryString, param)
+  table.insert(args, 1, req.queryString)
+  table.insert(args, 1, param)
+  args.n = 2
+  if not value then
+    return false
+  end
+  return true, {value}
+end
+say:set("assertion.req_query_param.negative", [[
+Expected query parameter: 
+%s
+But it was not found in: 
+%s
+]])
+say:set("assertion.req_query_param.positive", [[
+Did not expected query parameter: 
+%s
+But it was found in: 
+%s
+]])
+luassert:register("assertion", "queryparam", req_query_param,
+                  "assertion.req_query_param.negative",
+                  "assertion.req_query_param.positive")
+
+---
+-- Adds an assertion to look for a urlencoded form parameter in a `postData.params` subtable.
+-- Parameter name comparison is done case-insensitive.
+-- @return value of the parameter
+local function req_form_param(state, args)
+  local param = args[1]
+  local req = args[2] or kong_state.kong_request
+  if (not req) and kong_state.kong_request then
+    table.insert(args, 1, "<< 'formparam' assertion only works with a request object >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  if (type(req) ~= "table") or (type(req.postData) ~= "table") or 
+     (type(req.postData.params) ~= "table") then
+    table.insert(args, 1, "<< 'formparam' assertion input does not contain a 'postData.params' subtable >>")
+    table.insert(args, 1, param)
+    args.n = 2
+    return false
+  end
+  assert(req.postData.mimeType:lower():find("form-urlencoded",nil,true), 
+    "'formparam' assertion didn't get urlencoded data but; "..tostring(req.postData.mimeType))
+  local value = lookup(req.postData.params, param)
+  table.insert(args, 1, req.postData)
+  table.insert(args, 1, param)
+  args.n = 2
+  if not value then
+    return false
+  end
+  return true, {value}
+end
+say:set("assertion.req_form_param.negative", [[
+Expected url encoded form parameter: 
+%s
+But it was not found in: 
+%s
+]])
+say:set("assertion.req_form_param.positive", [[
+Did not expected url encoded form parameter: 
+%s
+But it was found in: 
+%s
+]])
+luassert:register("assertion", "formparam", req_form_param,
+                  "assertion.req_form_param.negative",
+                  "assertion.req_form_param.positive")
```

Comment:
Why do we need this assertion? I don't believe any test will have a use for it right now nor in the foreseeable future, right?